controllers/cloud.redhat.com/providers/kafka/strimzi.go

```diff
@@ -77,18 +81,24 @@ func (s *strimziProvider) configureKafkaCluster() error {
 
 	k.Spec = &strimzi.KafkaSpec{
 		Kafka: strimzi.KafkaSpecKafka{
+			Authorization: &strimzi.KafkaSpecKafkaAuthorization{
```

Comment:
Does this mean that apps must use auth to connect to kafka when using strimzi mode?

Reply:
My understanding of the code here was the app config will have both `brokers` and `brokersTLS` so if you want TLS you'd have to choose one of the TLS brokers.

Reply:
auth is orthogonal to tls

Reply:
unless there's an implicit lack of auth on non-tls connections

Reply:
So yeah, I see what you mean, if this config here sets the auth type globally for both tcp and TLS that's probably not ideal. We'll need to double check the strimzi docs...

Reply:
My hope is, this sets the auth type and applies only for a listener that has `Authentication` config on it

Reply:
https://strimzi.io/docs/operators/in-development/using.html#con-securing-kafka-authorization-str

> You can configure authorization for Kafka brokers using the authorization property in the Kafka.spec.kafka resource. If the authorization property is missing, no authorization is enabled and clients have no restrictions. When enabled, authorization is applied to **all enabled listeners**. The authorization method is defined in the type field.

Unless we intend to always use auth, it seems like we might need a flag in the kafka ClowdEnvironment config that allows us to turn auth completely off for a cluster?

If we always use auth, we'll have to modify various iqe plugins so they connect to kafka using users as well... we might want to coordinate on "flipping that switch"

Reply:
The Non TLS endpoint is not auth
The TLS endpoint is

Reply:
oh - perhaps it IS applied to all then

Reply:
I'll check this out - I hoped you had run it up @bsquizz as you have more kafka experience than I - I'll do this later

Reply:
Hah, yeah, I made the assumption auth wouldn't apply to a tcp listener... thinking "who would pass auth over a plain connection"? Based on the strimzi doc, bad assumption perhaps. There's also the chance the strimzi doc isn't 100% clear on it. Maybe we just need to spin it up and try it to confirm the behavior.

Reply:
yeh I can do that

Reply:
It is right - auth on means auth for all